File: weektwo/Sorting/Insertion.java
import java.util.Arrays;

public class Insertion {
    public static void main(String[] args) {
        int[] arr = { 3, 4, 2, 1, 5 };
        sort(arr);
        System.out.println(Arrays.toString(arr));
    }

    private static void sort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = i + 1; j > 0; j--) {
                if (arr[j] < arr[j - 1]) {
                    swap(j, j - 1, arr);
                } else {
                    break;
                }
            }
        }
    }

    private static void swap(int j, int j2, int[] arr) {
        int temp = arr[j];
        arr[j] = arr[j2];
        arr[j2] = temp;
    }
}

File: dsaJavascript/workout/sorting/insertion.js
function insertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
    for (let j = i; j > 0; j--) {
      if (arr[j] < arr[j - 1]) {
        swap(j, j - 1, arr);
      }else{
        break;
      }
    }
  }
}

function swap(i, j, arr) {
  let temp = arr[i];
  arr[i] = arr[j];
  arr[j] = temp;
}

let arr = [95, 2, 8, 99, 23, 1];
insertionSort(arr);

console.log(arr);
